### PR TITLE
Update ceph csi release 3.5 image and sidecars 

### PR DESCRIPTION
**Pull request by humblec**
Ceph CSI image has been udpated to v3.5.0
    
Below sidecars are updated to the latest releases with this commit.
    
    csi-provisioner      : 3.1.0
    csi-attacher         : 3.4.0
    node-driver-registrar: 2.4.0
    
    Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: Documentation/ceph-upgrade.md
Comment (Rakshith-R) [inline Documentation/ceph-upgrade.md:575]: change sidecar image versions here too

**Comment by Rakshith-R** [comment]
@humblec changes are required here too https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L32

**Comment by humblec** [comment]
> @humblec changes are required here too https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L32

@Rakshith-R added this too. ptal.. thanks!

**Review by leseb — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

File: pkg/operator/ceph/csi/version.go
Comment (Madhu-1) [inline pkg/operator/ceph/csi/version.go:31]: remove support for deprecated  3.3.0 release?

File: pkg/operator/ceph/csi/version_test.go
Comment (Madhu-1) [inline pkg/operator/ceph/csi/version_test.go:54]: this should be adjusted to 3.4.0 release as 3.3.0 is deprecated in cephcsi.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

**Comment by humblec** [comment]
@Madhu-1 iic, in previous discussions here in this repor, it was told that, *not* to deprecate the release in minor version of rook ? thats why 3.3.0 was kept still as minimum version, if we are fine to deprecate it for rook users too, I can adjust the PR... @travisn @leseb

**Comment by Madhu-1** [comment]
> @Madhu-1 iic, in previous discussions here in this repo, it was told that, _not_ to deprecate the release in minor version of rook ? thats why 3.3.0 was kept still as minimum version, if we are fine to deprecate it for rook users too, I can adjust the PR... @travisn @leseb

If the release is already deprecated in cephcsi what is the point of supporting an unsupported cephcsi version in Rook?

File: deploy/charts/rook-ceph/values.yaml
Comment (humblec) [reply deploy/charts/rook-ceph/values.yaml:283]: deliberately kept snapshotter aside as its not updated for ceph csi release too. May be along with resizer , it can be updated. I expect the resizer to land soon.

Comment (Madhu-1) [reply deploy/charts/rook-ceph/values.yaml:283]: are we going to wait till the resizer is released to merge this PR? if not IMO as the snapshotter release is done 18 days back it can be updated as part of this PR.

Comment (humblec) [reply deploy/charts/rook-ceph/values.yaml:283]: No.. we are not waiting.. but snapshotter is not tested against 3.5 release of ceph csi and there were some breaking changes in snapshotter 5.0.0 release too. so it has to be accomodated first in ceph csi to be in comfortable position here. so I would keep aside the snapshotter update till then.

**Comment by Madhu-1** [comment]
> > > @Madhu-1 iic, in previous discussions here in this repo, it was told that, _not_ to deprecate the release in minor version of rook ? thats why 3.3.0 was kept still as minimum version, if we are fine to deprecate it for rook users too, I can adjust the PR... @travisn @leseb
> > 
> > 
> > If the release is already deprecated in cephcsi what is the point of supporting an unsupported cephcsi version in Rook?
> 
> both repos deprecation policy is/can be different?

Here is what looks correct to me. If it's supported in cephcsi and deprecated in Rook then it's fine. but here it's different. If someone uses the older version and hits any bug, we can not do any bug fixes in cephcsi as it's already deprecated.

**Comment by humblec** [comment]
> > > > @Madhu-1 iic, in previous discussions here in this repo, it was told that, _not_ to deprecate the release in minor version of rook ? thats why 3.3.0 was kept still as minimum version, if we are fine to deprecate it for rook users too, I can adjust the PR... @travisn @leseb
> > > 
> > > 
> > > If the release is already deprecated in cephcsi what is the point of supporting an unsupported cephcsi version in Rook?
> > 
> > 
> > both repos deprecation policy is/can be different?
> 
> Here is what looks correct to me. If it's supported in cephcsi and deprecated in Rook then it's fine. but here it's different. If someone uses the older version and hits any bug, we can not do any bug fixes in cephcsi as it's already deprecated.

I am fine either way, will wait for @travisn and @leseb to confirm and can refresh the PR accordingly..

**Comment by travisn** [comment]
> > > > > @Madhu-1 iic, in previous discussions here in this repo, it was told that, _not_ to deprecate the release in minor version of rook ? thats why 3.3.0 was kept still as minimum version, if we are fine to deprecate it for rook users too, I can adjust the PR... @travisn @leseb
> > > > 
> > > > 
> > > > If the release is already deprecated in cephcsi what is the point of supporting an unsupported cephcsi version in Rook?
> > > 
> > > 
> > > both repos deprecation policy is/can be different?
> > 
> > 
> > Here is what looks correct to me. If it's supported in cephcsi and deprecated in Rook then it's fine. but here it's different. If someone uses the older version and hits any bug, we can not do any bug fixes in cephcsi as it's already deprecated.
> 
> I am fine either way, will wait for @travisn and @leseb to confirm and can refresh the PR accordingly..

If 3.3 is deprecated in CSI, it sounds fine to change the minimum version in rook to 3.4. If they still really want to run an older version, they can still override it with the use unsupported flag. But almost everyone upstream (I hope) uses the default csi version that comes with Rook, so I don't anticipate it will affect many users either.

**Comment by humblec** [comment]
> > 
> > I am fine either way, will wait for @travisn and @leseb to confirm and can refresh the PR accordingly..
> 
> If 3.3 is deprecated in CSI, it sounds fine to change the minimum version in rook to 3.4. If they still really want to run an older version, they can still override it with the use unsupported flag. But almost everyone upstream (I hope) uses the default csi version that comes with Rook, so I don't anticipate it will affect many users either.

Done.. :+1: , ptal @travisn

**Review by travisn — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**
